lib/catalog/update_order_item.rb

```diff
@@ -1,32 +1,22 @@
 module Catalog
   class UpdateOrderItem
-    def initialize(topic, task)
-      @payload = topic.payload
-      @message = topic.message
-      @task    = task
+    def initialize(topic, task, order_item)
+      @payload    = topic.payload
+      @message    = topic.message
+      @task       = task
+      @order_item = order_item
     end
 
     def process
       Rails.logger.info("Processing service order topic message: #{@message} with payload: #{@payload}")
 
-      Rails.logger.info("Searching for OrderItem with a task_ref: #{@payload["task_id"]}")
-      @order_item = find_order_item
-      Rails.logger.info("Found OrderItem: #{@order_item.id}")
-
       @order_item.update_message("info", "Task update message received with payload: #{@payload}")
 
       mark_item_based_on_status
     end
 
     private
 
-    def find_order_item
-      OrderItem.find_by!(:topology_task_ref => @payload["task_id"])
-    rescue ActiveRecord::RecordNotFound
```

Comment:
No sure we need to rescue this error here since we already validate the order contains a valid task_ref.

Reply:
I'm confused, are you saying we *should* rescue here? I removed the rescue because by passing the order item directly through the initializer there's no way we can find the wrong one, haha.

Reply:
My recommendation is not to pass in `order_item` directly. Then `find_order_item` is still needed, but no need to rescue the error.

The reason not to take `order_item` from initialization is to avoid taking an order and a task that are completely irrelevant. 

We don't need to rescue the error because through a normal procedure the task_id is already validated. If by any reason it is not, then let it naturally fail which can help discover the problem.